src/main/java/de/qabel/desktop/ui/AbstractController.java

```diff
@@ -55,4 +69,91 @@ private String getTraceAsString(Exception e) {
 			return null;
 		};
 	}
+
+	protected GsonContact createGsonFromEntity(Entity entity) {
+		GsonContact gc = new GsonContact();
+
+		if (entity instanceof Contact) {
+			Contact c = (Contact) entity;
+			gc.setEmail(c.getEmail());
+			gc.setPhone(c.getPhone());
+			gc.setAlias(c.getAlias());
+		} else {
+			Identity i = (Identity) entity;
+			gc.setEmail(i.getEmail());
+			gc.setPhone(i.getPhone());
+			gc.setAlias(i.getAlias());
+		}
+
+		gc.setCreated(entity.getCreated());
+		gc.setUpdated(entity.getUpdated());
+		gc.setDeleted(entity.getDeleted());
+		gc.setPublicKey(entity.getEcPublicKey().getKey());
+		for (DropURL d : entity.getDropUrls()) {
+			gc.addDropUrl(d.getUri().toString());
+		}
+		return gc;
+	}
+
+
+	protected void writeJsonInFile(String json, File dir) throws IOException {
+		InputStream stream = new ByteArrayInputStream(json.getBytes(StandardCharsets.UTF_8));
+		byte[] buffer = new byte[stream.available()];
+		stream.read(buffer);
+
+		File targetFile = new File(dir.getPath());
+		targetFile.createNewFile();
+		OutputStream outStream = new FileOutputStream(targetFile);
+		outStream.write(buffer);
+	}
+
+	protected void buildGson() {
+		final GsonBuilder builder = new GsonBuilder();
+		builder.serializeNulls();
+		builder.excludeFieldsWithoutExposeAnnotation();
+		gson = builder.create();
+	}
+
+	public String readFile(File f) throws IOException {
+		FileReader fileReader = new FileReader(f);
+		BufferedReader br = new BufferedReader(fileReader);
+
+		try {
```

Comment:
*hint:* with
```JAVA
try (StringBuilder sb = new StringBuilder()) {
 ...
}
```
br.close is called automatically at the end of the try-block and the finally is implicit